tests/test_discrete_quadratic_model.py

```diff
@@ -409,6 +409,29 @@ def test_set_quadratic_offdtype(self, name, dtype):
         dqm.set_quadratic(u, v, biases)
 
 
+class TestConstraint(unittest.TestCase):
+    def test_simple_constraint(self):
+        dqm = dimod.DQM()
+        num_variables = 2
+        num_cases = 3
+        x = {}
+        for i in range(num_variables):
+            x[i] = dqm.add_variable(num_cases, label=f'x_{i}')
```

Comment:
this line will cause issues in python 3.5 because f strings were not supported at the time. I think instead you could do `label='x_{i}'.format(i=i)`